### ISSUE TITLE
How to prevent panel actions collapse?

### ISSUE THREAD

**Issue by inwardmovement** [iocave/monkey-patch#13]
How to prevent panel actions collapse?
It seems to be related to 
```
https://ticino.blob.core.windows.net/sourcemaps/17299e413d5590b14ab0340ea477cdd86ff13daf/core/vs/workbench/vs/workbench/file:/__w/1/s/src/vs/base/browser/ui/actionbar/actionbar.ts
```

```
	pull(index: number): void {
		if (index >= 0 && index < this.viewItems.length) {
			this.actionsList.removeChild(this.actionsList.childNodes[index]);
			dispose(this.viewItems.splice(index, 1));
		}
	}
```

![vsc](https://user-images.githubusercontent.com/9438102/87871360-00055d00-c9b0-11ea-936f-7e2096621954.gif)

**Comment by knopp** [comment]
That's a bit weird. When I move panel weird the search action icon are to the right of "tabs", in your case they seem to be to the bottom for some reason

**Comment by inwardmovement** [comment]
Yes because I moved them with the Customize UI extension.

Le dim. 19 juil. 2020 à 13:52, Matej Knopp <notifications@github.com> a
écrit :

**Comment by knopp** [comment]
CustomizeUI currently doesn't have any feature for moving these icons, how exactly did you move them? Also this issue should have been opened for customize-ui extension.

**Comment by inwardmovement** [comment]
> how exactly did you move them?

```
"customizeUI.stylesheet": {
    ".panel .title": "flex-direction: column !important; height: 65px !important;",
    ".panel .title .actions-container": "justify-content: flex-start !important;"
  }
```

But it's not the point here.

At first sight I thought like you, but then I realized that this behavior is not made with CSS but with JS as I explain in my first message, so I guessed it should be customized with Monkey Patch, not Customize UI.

**Comment by knopp** [comment]
Right. This is not going to be enough. Customizing layout usually needs monkey patching layout code, which is lot more involved. You can look at how customize UI does it, it can get quite tricky. It involves looking at original vscode source code, try to find good replacement points and then replacing implementations with own layout functions.
